### PR TITLE
[JENKINS-31047] Incorrect Behaviour on WarningsPluginTest

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
[JENKINS-31047](https://issues.jenkins-ci.org/browse/JENKINS-31047)

@reviewbybees

### CODE REVIEW

**Comment by andresrc** [comment]
:bee:

**Comment by Vlatombe** [comment]
🐝

**Comment by armfergom** [comment]
Seems reasonable. 🐝 provided it works

**Comment by raul-arabaolaza** [comment]
@armfergom Check the CI build:

> Running plugins.WarningsPluginTest
> /usr/bin/curl
> Tests run: 23, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2,626.247 sec - in plugins.WarningsPluginTest

Is working, so I am going to mark it as reviewed

@reviewbybees done

**Comment by recena** [comment]
@abayer @olivergondza Do you agree with it? Maybe merge?

File: src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java [hunk 1]
```diff
@@ -14,4 +15,28 @@
...
+
+    @Override
+    public <T extends PostBuildStep> T addPublisher(Class<T> type) {
+        T p = addStep(type, "publisher");
+
+        publishers.add(p);
+        return p;
+    }
+
+    private <T extends Step> T addStep(Class<T> type, String section) {
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java]: Making `Job#addStep` protected should make this fix simpler.

File: src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java [hunk 2]
```diff
@@ -14,4 +15,28 @@
...
+        ensureConfigPage();
+
+        control(by.path("/projectFactory/hetero-list-add[%s]", section)).selectDropdownMenu(type);
+        waitFor(by.xpath("//div[@name='%s']", section), 3); // it takes some time until the element is visible
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java]: IIUC stretching the timeout is the only reason this override was introduced. Why not make this improvement on `Job`?

Comment (raul-arabaolaza) [inline src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java]: @olivergondza No, it wasn't the only reason, but you are right in your both comments, I will create a follow up PR with your suggestions

File: src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java (hunk 1 above)
Comment (raul-arabaolaza) [inline src/main/java/org/jenkinsci/test/acceptance/po/FreeStyleMultiBranchJob.java]: @olivergondza Unfortunately the FreeStyleMultiBranchJob' s UI mixes the standard hetero-list with another one with other path only for publishers and builder, so overriding the `Job#addStep` method results in other parts of the test failing, I need to separate methods

**Pull request merged**